contracts/javascore/cluster-connection/src/test/java/xcall/adapter/cluster/ClusterConnectionTest.java

```diff
@@ -0,0 +1,270 @@
+    }
+
+    @Test
+    public void testRevertMessage() {
+
+        connection.invoke(source_relayer, "revertMessage", BigInteger.ONE);
+    }
+
+    @Test
+    public void testRevertMessage_unauthorized(){
+        UserRevertedException e = assertThrows(UserRevertedException.class, ()->connection.invoke(user, "revertMessage", BigInteger.ONE));
+        assertEquals("Reverted(0): "+"Only relayer can call this function", e.getMessage());
+        
+    }
+
+    @Test
+    public void testSetFeesUnauthorized(){
+        UserRevertedException e = assertThrows(UserRevertedException.class,() -> connection.invoke(user, "setFee", "0xevm",
+        BigInteger.TEN, BigInteger.TEN));
+        assertEquals("Reverted(0): "+"Only relayer can call this function", e.getMessage());
+    }
+
+    @Test
+    public void testClaimFees(){
+        setFee();
+        connection.invoke(source_relayer, "claimFees");
+        assertEquals(source_relayer.getBalance(), BigInteger.ZERO);
+
+        UserRevertedException e = assertThrows(UserRevertedException.class,() -> connection.invoke(callservice.account, "sendMessage", nidTarget,
+        "xcall", BigInteger.ONE, "null".getBytes()));
+        assertEquals(e.getMessage(), "Reverted(0): Insufficient balance");
+
+        try (MockedStatic<Context> contextMock = Mockito.mockStatic(Context.class, Mockito.CALLS_REAL_METHODS)) {
+            contextMock.when(() -> Context.getValue()).thenReturn(BigInteger.valueOf(20));
+            connection.invoke(callservice.account, "sendMessage", nidTarget,"xcall", BigInteger.ONE, "null".getBytes());
+        }
+
+        
+        try (MockedStatic<Context> contextMock = Mockito.mockStatic(Context.class, Mockito.CALLS_REAL_METHODS)) {
+            contextMock.when(() -> Context.getBalance(connection.getAddress())).thenReturn(BigInteger.valueOf(20));
+            contextMock.when(() -> Context.transfer(source_relayer.getAddress(),BigInteger.valueOf(20))).then(invocationOnMock -> null);
+            connection.invoke(source_relayer, "claimFees");
+        }
+    }
+
+    @Test
+    public void testClaimFees_unauthorized(){
+        setFee();
+        UserRevertedException e = assertThrows(UserRevertedException.class,() -> connection.invoke(user, "claimFees"));
+        assertEquals(e.getMessage(), "Reverted(0): "+"Only relayer can call this function");
+    }
+
+    public MockedStatic.Verification value() {
+        return () -> Context.getValue();
+    }
+
+    @Test
+    public void testGetReceipt(){
+        assertEquals(connection.call("getReceipts", nidSource, BigInteger.ONE),
+        false);
+
+        connection.invoke(source_relayer, "recvMessage",nidSource, BigInteger.ONE, "test".getBytes());
+
+        assertEquals(connection.call("getReceipts", nidSource, BigInteger.ONE),
+        true);
+    }
+
+    @Test
+    public void testRecvMessageWithSignatures() throws Exception{
+        byte[] data = "test".getBytes();
```

Comment:
it would make sense to get some sample data from relayer and test it here , to make sure everything is in sync.